source/main/modules/window-manager/index.ts

```diff
@@ -207,18 +223,72 @@ export default class WindowManager {
   /**
    * Shuts down the window manager and performs final operations
    */
-  shutdown (): void {
+  shutdown(): void {
     this._persistWindowPositions()
   }
 
   /**
    * Listens to events on the main window
    */
-  private _hookMainWindow (): void {
+  private _hookMainWindow(): void {
     if (this._mainWindow === null) {
       return
     }
 
+    this._mainWindow.on('show', async () => {
+      if (process.platform == 'darwin') {
+        if (tray == null) {
```

Comment:
Can the tray creation be put in a function? Because in #33 the tray must be created after the preference is set to true. The tray creation function will be called externally from `source/win-preferences/preferences.vue`.